### ISSUE TITLE
Crash

### ISSUE THREAD

**Issue by Serega007RU** [xxDark/BetterLoading#2]
[crash-2021-10-12_20.58.09-client.txt](https://github.com/xxDark/BetterLoading/files/7332212/crash-2021-10-12_20.58.09-client.txt)
[latest.log](https://github.com/xxDark/BetterLoading/files/7332214/latest.log)

**Comment by xxDark** [comment]
For some reason, GitHub decided not to mention me. Is this still an issue?

**Comment by Serega007RU** [comment]
Yes, this problem is still relevant

**Comment by xxDark** [comment]
I made the change that should probably fix the issue, but I currently have no time to confirm that. Sorry for taking too long to respond.

**Comment by Serega007RU** [comment]
Your change fixed the crash, but now minecraft loads endlessly. I noticed that the mod still loads without the fabric-api, but unfortunately the fabric-api is needed for most mods.
![image](https://user-images.githubusercontent.com/64780091/140819143-d24fa6fe-1b8f-4188-8fa5-6e711688dfbd.png)
[2021-11-08-7.log.gz](https://github.com/xxDark/BetterLoading/files/7500182/2021-11-08-7.log.gz)
Here is the log with deobfuscation
[2021-11-09-1.log.gz](https://github.com/xxDark/BetterLoading/files/7500197/2021-11-09-1.log.gz)

**Comment by xxDark** [comment]
> 
> 
> Your change fixed the crash, but now minecraft loads endlessly. I noticed that the mod still loads without the fabric-api, but unfortunately the fabric-api is needed for most mods. ![image](https://user-images.githubusercontent.com/64780091/140819143-d24fa6fe-1b8f-4188-8fa5-6e711688dfbd.png) [2021-11-08-7.log.gz](https://github.com/xxDark/BetterLoading/files/7500182/2021-11-08-7.log.gz) Here is the log with deobfuscation [2021-11-09-1.log.gz](https://github.com/xxDark/BetterLoading/files/7500197/2021-11-09-1.log.gz)

Yeah, I did not manage to solve that conflict, unfortunately, see https://github.com/FabricMC/fabric/issues/1733

**Comment by xxDark** [comment]
I might be able to create a workaround for that, however, I'm not sure if it will affect other mods that were relying on a such injection in a fabric itself.

**Comment by xxDark** [comment]
I pushed the commit that might solve your issue

**Comment by Serega007RU** [comment]
Now minecraft crashes, right at startup
[crash-2021-11-10_04.42.30-client.txt](https://github.com/xxDark/BetterLoading/files/7509177/crash-2021-11-10_04.42.30-client.txt)

**Comment by xxDark** [comment]
> 
> 
> Now minecraft crashes, right at startup [crash-2021-11-10_04.42.30-client.txt](https://github.com/xxDark/BetterLoading/files/7509177/crash-2021-11-10_04.42.30-client.txt)

I don't know what kind of mixins Fabric is using, but this is really strange.
Should be fixed

**Comment by Serega007RU** [comment]
There are no errors in the logs now, but minecraft loads endlessly, watch the video
https://user-images.githubusercontent.com/64780091/141039567-f7fb82f1-a892-48c2-bfeb-c68c8d6260b5.mp4
[2021-11-10-5.log.gz](https://github.com/xxDark/BetterLoading/files/7509408/2021-11-10-5.log.gz)

**Comment by xxDark** [comment]
Works fine for me tho. Can you send me the mods you are using?

**Comment by Serega007RU** [comment]
The video and logs show which mods I use (in general, I deleted all my mods for the purity of the experiment and left only the main ones)
Fabric-API: https://www.curseforge.com/minecraft/mc-mods/fabric-api/files/all?filter-game-version=2020709689%3A8516
Needed for most mods
I also use Not Enough Crashes there, but minecraft also loads endlessly without this mod


Let's speak Russian? You seem to be Russian, my English is bad

**Comment by xxDark** [comment]
Можно. У меня установлены только эти моды, и всех остальных нет
![img](https://user-images.githubusercontent.com/19853368/141108920-f76c1b47-6766-4a00-bd9a-6a5d028a6c65.png)
Скинь все файлы, так будет проще

**Comment by Serega007RU** [comment]
[mods.zip](https://github.com/xxDark/BetterLoading/files/7512582/mods.zip)

**Comment by xxDark** [comment]
Деситно сыплет ошибками, буду разбираться

**Comment by xxDark** [comment]
Код в фабрике ломает оригинальное поведение ваниллы, открою у них issue

**Comment by xxDark** [comment]
Исправлено

**Comment by xxDark** [comment]
@Serega007RU Еще можешь ради интереса сравнить время запуска? Хотя бы по логам. У меня на моей пекарне это приличный буст дает, но хотелось бы конечно знать, что не мне одному.

**Comment by Serega007RU** [comment]
Minecraft всё также бесконечно загружается как на том видео
[2021-11-11-4.log.gz](https://github.com/xxDark/BetterLoading/files/7517548/2021-11-11-4.log.gz)
Может это ещё зависит от версии Fabric-Loader? У меня она последняя, 0.12.5, обновить её можно тут через установщик https://fabricmc.net/use/

**Comment by xxDark** [comment]
> 
> 
> Minecraft всё также бесконечно загружается как на том видео [2021-11-11-4.log.gz](https://github.com/xxDark/BetterLoading/files/7517548/2021-11-11-4.log.gz) Может это ещё зависит от версии Fabric-Loader? У меня она последняя, 0.12.5, обновить её можно тут через установщик https://fabricmc.net/use/

Добавил пару проверок, весело, конечно..
На 0.17 все прекрасно, на 0.12.5 все сломано.
![crash](https://user-images.githubusercontent.com/19853368/141281110-a9085506-3076-4e9a-bd67-5c974ca26dc4.png)

UPD: Исправил

**Comment by Serega007RU** [comment]
Ну наконец ничего не крашится :D
Теперь следующая проблема, текстур пак Programmer Art перестал работать хотя если ставить свои текстур паки - они работают
Я пробовал его и выключать и обратно включать, ничего не помогает

**Comment by xxDark** [comment]
Посмотрю, как будет время

**Comment by xxDark** [comment]
Проверяй.

**Comment by Serega007RU** [comment]
Всё также, не помогло
